pyarr/sonarr_api.py

```diff
@@ -1,470 +1,677 @@
+
+    # PUT /episodefile/{id}
+    def upd_episode_file_quality(self, id_, data):
+        """Updates the quality of the episode file and returns the episode file.
+
+        Args:
+            [Required] id_ (str)
+            [Required] data (dict) - See Sonarr docs for formatting
+
+        Returns:
+            JSON Response (dict)
+        """
+        path = f"/api/episodefile/{id_}"
+        res = self.request_put(path, data=data)
+        return res
+
+    ## HISTORY
+
+    # GET /history
+    def get_history(self, sort_key, page=1, page_size=10, sort_dir="desc", id_=None):
+        """Gets history (grabs/failures/completed)
+
+        Args:
+            [Required] sort_key (str) - series.title or date
+            [Optional] page (int) - Default: 1
+            [Optional] page_size (int) - Default: 10
+            [Optional] sort_dir (str) - asc or desc - Default: desc
+            [Optional] id_ (int) - Filters to a specific episode ID
+        Returns:
+            JSON Response (dict)
+        """
+        path = "/api/history"
+        params = {
+            "sortKey": sort_key,
+            "page": page,
+            "pageSize": page_size,
+            "sortDir": sort_dir,
+        }
+        if id_:
+            params["episodeId"] = id_
+        res = self.request_get(path, params=params)
+        return res
+
+    ## WANTED (MISSING)
+
+    # GET /wanted/missing
+    def get_wanted(self, sort_key, page=1, page_size=10, sort_dir="asc"):
+        """Gets Wanted / Missing episodes
+
+        Args:
+            [Required] sort_key (str) - series.title or airDateUtc
+            [Optional] page (int) - Default: 1
+            [Optional] page_size (int) - Default: 10
+            [Optional] sort_dir (str) - asc or desc - Default: asc
+        Returns:
+            JSON Response (dict)
+        """
+        path = "/api/wanted/missing"
+        params = {
+            "sortKey": sort_key,
+            "page": page,
+            "pageSize": page_size,
+            "sortDir": sort_dir,
+        }
+        res = self.request_get(path, params=params)
+        return res
+
+    ## QUEUE
+
+    # GET /queue
+    def get_queue(self):
+        """Gets current downloading info
+
+        Args:
+            None
+        Returns:
+            JSON Response (dict)
+        """
+        path = "/api/queue"
         res = self.request_get(path)
         return res
 
+    # DELETE /queue
+    def del_queue(self, id_, blacklist=True):
+        """Deletes an item from the queue and download client.
+        Optionally blacklist item after deletion.
+
+        Args:
+            [Required] id_ (int)
+            [Optional] blacklist (bool) - Default: True
+        Returns:
+            JSON Response (dict)
+        """
+        params = {"id": id_, "blacklist": blacklist}
+        path = "/api/queue/"
+        res = self.request_del(path, params=params)
+        return res
+
+    ## PARSE
+
+    # GET /parse
+    def get_parsed_title(self, title):
```

Comment:
Error is displayed if a title is provided:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/marksie1988/repos/pyarr/pyarr/sonarr_api.py", line 335, in get_parsed_title
    res = self.request_get(path, params=params)
  File "/home/marksie1988/repos/pyarr/pyarr/request_api.py", line 51, in request_get
    return res.json()
  File "/usr/lib/python3/dist-packages/requests/models.py", line 898, in json
    return complexjson.loads(self.text, **kwargs)
  File "/usr/lib/python3/dist-packages/simplejson/__init__.py", line 525, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3/dist-packages/simplejson/decoder.py", line 370, in decode
    obj, end = self.raw_decode(s)
  File "/usr/lib/python3/dist-packages/simplejson/decoder.py", line 400, in raw_decode
    return self.scan_once(s, idx=_w(s, idx).end())
simplejson.errors.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

```

Reply:
Unknown why this one fails - I'll need to test this in the future.

Reply:
yes using the fill title as `episodetitle.S01E01.HDTV-720p` does work